---
title: Gestion des Erreurs et Exceptions
---

La robustesse d'un programme r√©side souvent dans sa capacit√© √† g√©rer les impr√©vus. Python fournit des outils puissants pour traiter et anticiper les erreurs √† travers le m√©canisme d'exceptions.

#### **Comprendre les Erreurs et Exceptions** üí°

- **Erreur (Bug)**: Il s'agit d'un d√©faut dans le code. Par exemple, une variable non d√©finie.
  
- **Exception**: C'est un √©v√©nement qui se produit pendant l'ex√©cution du programme et qui interrompt le flux normal d'instructions.

#### **Blocs Try, Except, Else et Finally** üõë

- **Bloc `try`**: Contient le code susceptible de g√©n√©rer une exception.
  
- **Bloc `except`**: G√®re l'exception si elle est lev√©e dans le bloc `try`.
  
- **Bloc `else`**: Ce bloc est ex√©cut√© si aucune exception n'est lev√©e dans le bloc `try`.
  
- **Bloc `finally`**: Ex√©cut√© toujours √† la fin, qu'une exception soit lev√©e ou non.

```python
try:
    resultat = int(input("Entrez un nombre: ")) / 2
except ValueError:
    print("Veuillez entrer un nombre valide!")
else:
    print(f"La moiti√© de votre nombre est {resultat}.")
finally:
    print("Merci d'avoir particip√©!")
```

#### **Types courants d'Exceptions** üìå

Python poss√®de de nombreuses exceptions int√©gr√©es comme:

- `ValueError`: Lorsqu'une fonction re√ßoit un argument de valeur correcte mais de type inappropri√©.
- `TypeError`: Lorsque une op√©ration est appliqu√©e au type inappropri√©.
- `FileNotFoundError`: Lorsqu'un fichier ou un r√©pertoire est demand√© mais n'existe pas.

#### **Lev√©e d'Exceptions** üöÄ

Le mot-cl√© `raise` permet de lancer une exception sp√©cifique, ce qui peut √™tre utile pour signaler des erreurs dans votre propre code.

```python
age = int(input("Quel est votre √¢ge? "))
if age < 0:
    raise ValueError("L'√¢ge ne peut pas √™tre n√©gatif!")
```

#### **Assertions** ‚õî

Les assertions sont un outil de d√©bogage, elles testent une condition et d√©clenchent une exception si la condition est fausse.

```python
age = int(input("Entrez votre √¢ge: "))
assert age >= 0, "L'√¢ge ne peut pas √™tre n√©gatif!"
```

#### **Exceptions Personnalis√©es** üé®

La cr√©ation d'exceptions personnalis√©es peut aider √† clarifier votre code et √† traiter des cas sp√©cifiques.

```python
class AgeNegatif(Exception):
    """Exception lev√©e pour un √¢ge n√©gatif."""
    pass

age = int(input("Entrez votre √¢ge: "))
if age < 0:
    raise AgeNegatif("L'√¢ge ne peut pas √™tre n√©gatif!")
```

---

La gestion des erreurs est essentielle pour la robustesse d'un programme. La capacit√© de pr√©voir, g√©rer et communiquer les erreurs rend un programme plus fiable et plus convivial.